### DISCUSSION TITLE
How should the code generation be run locally?

### DISCUSSION THREAD

**Discussion by JonathanPlasse** [nutti/fake-bpy-module#165] [Q&A] [answered]
I am using the command below, is there a better way
```
bash tools/pip_package/build_pip_package.sh blender 3.3 ../../blender/blender ../../../../../opt/blender3.3.4 3.3
```
Could I use only this command?
```
python src/gen.py -- -i tools/pip_package/tmp-3.3/gen_module-tmp/sphinx-in -o out -f pep8 -T blender -t 3.3 -l warn -m 3.3
```

**Comment by nutti** [comment, marked as answer]
@JonathanPlasse 

There are 3 ways to generate modules.

* Run `src/gen.py`
  * This does not include the patches and mod files. You can not generate complete modules by this method.
* Run `src/gen_module.sh`
  * This script generates complete modules without packaging. I recommend you to use this script for the testing purpose.
* Run `tools/pip_package/build_pip_package.sh`
  * This script generates complete modules with packaging.

**Reply by JonathanPlasse** [reply]
Thanks, that is what I will use.
